flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/ddl/SqlCreateTable.java

```diff
@@ -189,6 +198,15 @@ public void validate() throws SqlValidateException {
         if (tableLike != null) {
             tableLike.validate();
         }
+
+        if (query != null
+                && (columnList.size() > 0
```

Comment:
I think we should validate these in separated if branch. In each case, we should give a clear error message.